src/core/iomgr/pollset_posix.c

```diff
@@ -247,8 +246,11 @@ static void finish_shutdown(grpc_exec_ctx *exec_ctx, grpc_pollset *pollset) {
 }
 
 void grpc_pollset_work(grpc_exec_ctx *exec_ctx, grpc_pollset *pollset,
-                       grpc_pollset_worker *worker, gpr_timespec now,
+                       grpc_pollset_worker **worker_hdl, gpr_timespec now,
                        gpr_timespec deadline) {
+  grpc_pollset_worker worker;
+  *worker_hdl = &worker;
```

Comment:
Again, I'm missing the point of the handle. I also looked at all the codes that use it and am still not any the wiser. The pointer starts off as NULL, gets passed into here where it gets pointed to a local variable, and then gets reset back to NULL before the function finishes. Is that about right?